pkg/controllers/nodepool/readiness/controller.go

```diff
@@ -54,20 +54,29 @@ func NewController(kubeClient client.Client, cloudProvider cloudprovider.CloudPr
 func (c *Controller) Reconcile(ctx context.Context, nodePool *v1.NodePool) (reconcile.Result, error) {
 	ctx = injection.WithControllerName(ctx, "nodepool.readiness")
 	stored := nodePool.DeepCopy()
-	supportedNC := c.cloudProvider.GetSupportedNodeClasses()
 
-	if len(supportedNC) == 0 {
-		logger.Fatal("no supported node classes found for the cloud provider")
+	nodeClass, ok := lo.Find(c.cloudProvider.GetSupportedNodeClasses(), func(nc status.Object) bool {
```

Comment:
We dropped our error checking on GetSupportedNodeClasses(). How do we dry-run to make sure that CloudProviders have defined this?

Reply:
Discussed offline, we could potentially do this in the operator but it's not really necessary. We'll punt for now.